web/hooks/github_test.go

```diff
@@ -76,26 +78,151 @@ func TestGitHubWebHook(t *testing.T) {
+func TestGitHubWebHookOrg(t *testing.T) {
+	qfTestOrg := scm.GetTestOrganization(t)
+	accessToken := scm.GetAccessToken(t)
+	serverURL := scm.GetWebHookServer(t)
+
+	logger := logq.Zap(true)
+	defer func() { _ = logger.Sync() }()
+
+	s, err := scm.NewSCMClient(logger.Sugar(), "github", accessToken)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx := context.Background()
+	opt := &scm.CreateHookOptions{
+		URL:          serverURL + "/webhook",
+		Secret:       secret,
+		Organization: qfTestOrg,
+	}
+	err = s.CreateHook(ctx, opt)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	course := &pb.Course{
+		Name:             qfTestOrg,
+		OrganizationPath: qfTestOrg,
+		Code:             qfTestOrg,
+		Provider:         "github",
+	}
+
+	db, cleanup := qtest.TestDB(t)
+	defer cleanup()
+	if err := qtest.PopulateDatabaseWithInitialData(t, db, s, course); err != nil {
+		t.Fatal(err)
+	}
+	if err := populateDatabaseWithTasks(t, ctx, logger.Sugar(), db, s, course); err != nil {
+		t.Fatal(err)
+	}
+	runner, err := ci.NewDockerCI(logger)
+	if err != nil {
+		logger.Sugar().Errorf("failed to set up docker client: %v\n", err)
+	}
+	defer runner.Close()
+	webhook := NewGitHubWebHook(logger.Sugar(), db, runner, secret)
+
+	log.Println("starting webhook server")
+	http.HandleFunc("/webhook", webhook.Handle)
+	log.Fatal(http.ListenAndServe(":8080", nil))
+}
+
+// TODO(Espeland): This function is almost identical to the one in assignments/tasks_test.go.
+// It would be best if we could add this to qtest, which is currently not possible since using FetchAssignments there would lead to an import cycle.
+func populateDatabaseWithTasks(t *testing.T, ctx context.Context, logger *zap.SugaredLogger, db database.Database, sc scm.SCM, course *pb.Course) error {
+	t.Helper()
+
+	type foundIssue struct {
+		IssueNumber uint64
+		Name        string
+	}
+
+	org, err := sc.GetOrganization(ctx, &scm.GetOrgOptions{Name: course.Name})
+	if err != nil {
+		return err
+	}
+
+	// Find and create assignments
+	foundAssignments, dockerfile, err := assignments.FetchAssignments(ctx, logger, sc, course)
+	if err != nil {
+		return err
+	}
+	course.Dockerfile = dockerfile
+	if err = db.UpdateCourse(course); err != nil {
+		return err
+	}
+
+	if err = db.UpdateAssignments(foundAssignments); err != nil {
+		return err
+	}
+
+	repos, err := sc.GetRepositories(ctx, org)
+	if err != nil {
+		return err
+	}
+
+	foundIssues := make(map[uint64]map[string]*foundIssue)
+	tasks := make(map[uint32]map[string]*pb.Task)
+
+	// Finds issues, and creates tasks based on them
+	for _, repo := range repos {
+		existingScmIssues, err := sc.GetIssues(ctx, &scm.RepositoryOptions{
+			Owner: course.Name,
+			Path:  repo.Path,
+		})
+		if err != nil {
+			return err
+		}
+
+		if len(existingScmIssues) == 0 {
+			continue
+		}
+		foundIssues[repo.ID] = make(map[string]*foundIssue)
+		for _, scmIssue := range existingScmIssues {
+			splitTitle := strings.Split(scmIssue.Title, ", ")
```

Comment:
I'm not a fan of this formatted string to get the assignmentOrder. Is this also used for the production version? For testing, it might be okay... although I would prefer a more robust approach.

Reply:
Again. I think this entire approach should be scrapped instead of fixed.

Reply:
Yes, it is only for testing